PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,109 @@
+        return new ConnectionStringVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * This class extends the JavaElementVisitor to visit the elements in the code.
+     * It checks if the method call is a connection string call and if the class is an Azure client.
+     * If both conditions are met, a problem is registered with the suggestion message.
+     */
+    public static class ConnectionStringVisitor extends JavaElementVisitor {
+
+        private final ProblemsHolder holder;
+        private final boolean isOnTheFly;
+        private static final Logger LOGGER = Logger.getLogger(ConnectionStringCheck.class.getName());
+        private static String METHOD_TO_CHECK = "";
+        private static String SUGGESTION = "";
+
+        // Load the JSON configuration file to get the method to check and the suggestion message
+        static {
+            String ruleName = "ConnectionStringCheck";
+
+            try {
+                LoadJsonConfigFile config = LoadJsonConfigFile.getInstance();
+                METHOD_TO_CHECK = config.getMethodToCheck(ruleName);
+                SUGGESTION = config.getSuggestionMessage(ruleName);
+            } catch (IOException e) {
+                // make a descriptive message
+                LOGGER.log(Level.SEVERE, "Failed to load JSON configuration for rule '" + ruleName + "' from expected path: " + LoadJsonConfigFile.CONFIG_FILE_PATH, e);
+            }
+        }
+
+        /**
+         * Constructor to initialize the visitor with the holder and isOnTheFly flag.
+         * @param holder     ProblemsHolder
+         * @param isOnTheFly boolean
+         */
+        public ConnectionStringVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        /**
+         * This method visits the element in the code.
+         * It checks if the method call is a connection string call and if the class is an Azure client.
+         * If both conditions are met, a problem is registered with the suggestion message.
+         * @param element PsiElement
+         */
+        @Override
+        public void visitElement(@NotNull PsiElement element) {
+            super.visitElement(element);
+
+            // check if an element is an azure client
+            if (element instanceof PsiMethodCallExpression) {
+
+                PsiMethodCallExpression methodCallExpression = (PsiMethodCallExpression) element;
+                PsiReferenceExpression methodExpression = methodCallExpression.getMethodExpression();
+
+                // resolvedMethod is the method that is being called
+                PsiElement resolvedMethod = methodExpression.resolve();
+
+                // check if the method is a connectionString call
+                if (resolvedMethod != null && resolvedMethod instanceof PsiElement && ((PsiMethod) resolvedMethod).getName().equals(METHOD_TO_CHECK)) {
```

Comment:
Why can't we directly check that resolvedMethod be an instance of PsiMethod insteaod PsiElement in the if check ?

Reply:
yehpp changed it to resolvedMethod instanceof PsiMethod